challenges/fizzBuzzTree/fizz-buzz-tree.js

```diff
@@ -0,0 +1,55 @@
+// Node
+class Node {
+  constructor(value, left = null, right = null) {
+    this.value = value
+    this.left = left
+    this.right = right
+  }
+}
+
+//Binary Tree class
+class BinaryTree {
+  constructor() {
+    this.root = null;
+  }
+
+  fizzBuzzTree() {
+    // make new tree
+    let newTree = new BinaryTree
+    newTree.root = this.root
+
+    // traverse tree
+    let traverse = (node) => {
+      if (node === null) {
+        return null
+      }
+
+      if (node.value%15 === 0) {
+        node.value = 'FizzBuzz'
+      } else if (node.value%3 === 0) {
+        node.value = 'Fizz'
+      } else if (node.value%5 === 0) {
+        node.value = 'Buzz'
+      } else {
+        node.value = node.value.toString()
+      }
```

Comment:
You changed the original tree. Can you come up with a solution that generate a new fizzbuzz tree?

Reply:
Based on the code you showed in class, Yes I think I can do that

Reply:
See pr https://github.com/astrokd/data-structures-and-algorithms/pull/46